build.gradle

```diff
@@ -26,28 +26,35 @@ ext.ref = parseprops(file('build.properties'))
 ext.ref.mod_deps = parseprops(file('mod_deps.properties'))
 
 group = ref.package_base + '.' + ref.mod_id
-archivesBaseName = ref.mod_abbr
-// remove travis stuff from version for now because we aren't using it and it was a mess. -srs
-version = ref.mc_version + '-' + ref.mod_version
+archivesBaseName = (ref.mod_id as String).toUpperCase() + '-' + (ref.mc_version as String)
+version = ref.mod_version
+
+repositories {
+    maven { url "http://files.minecraftforge.net/maven" } //BiomesOPlenty
+    maven { url 'https://dl.bintray.com/shinoow/maven/' } //AbyssalCraft
+}
+
+dependencies {
+    provided fileTree(dir: 'libs', include: '*.jar')
+    deobfProvided "com.github.glitchfiend.biomesoplenty:BiomesOPlenty:" + (ref.mod_deps.bopver as String) + ":deobf"
+    deobfProvided 'com.shinoow.abyssalcraft:AbyssalCraft:' + (ref.mod_deps.acver as String) + ':dev'
+}
 
 minecraft {
-    version = (ref.mcf_postfix!='') ? ref.mcf_version + '-' + ref.mcf_postfix : ref.mcf_version
+    version = (ref.mcf_suffix!='') ? ref.mcf_version + '-' + ref.mcf_suffix : ref.mcf_version
+    mappings = ref.mcp_mappings
     runDir = ref.run_dir
-    mappings = ref.mappings
     makeObfSourceJar = false
-    replace '@MOD_NAME@', ref.mod_name
-    replace '@MOD_ID@', ref.mod_id
-    replace '@MOD_VERSION@', version
-    replace '@MCF_VERSION@', ref.mcf_version
-    if (ref.mod_deps!="") replace '@MOD_DEPS@', ";" + ref.mod_deps.depstring
+    replace '@MOD_ID@',         ref.mod_id
+    replace '@MOD_NAME@',       ref.mod_name
+    replace '@MOD_VERSION@',    project.version
+    replace '@MCF_MINVER@',     ref.mcf_minver
```

Comment:
I'm getting a crash when trying to launch MC from within IDEA:

`java.lang.RuntimeException: Invalid versioned reference Forge@[@MCF_MINVER@,@MCF_MAXVER@)@MOD_DEPS@`

Anything we can do?